gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala

```diff
@@ -171,31 +170,30 @@ case class TransformPreOverrides() extends Rule[SparkPlan] {
             plan.resultExpressions,
             child)
       case plan: UnionExec =>
-        val children = plan.children.map(
-          replaceWithTransformerPlan(_, isSupportAdaptive))
+        val children = plan.children.map(replaceWithTransformerPlan)
         logDebug(s"Columnar Processing for ${plan.getClass} is currently supported.")
         UnionExecTransformer(children)
       case plan: CustomExpandExec =>
-        val child = replaceWithTransformerPlan(plan.child, isSupportAdaptive)
+        val child = replaceWithTransformerPlan(plan.child)
         logDebug(s"Columnar Processing for ${plan.getClass} is currently supported.")
         ExpandExecTransformer(plan.projections, plan.groupExpression, plan.output, child)
       case plan: SortExec =>
-        val child = replaceWithTransformerPlan(plan.child, isSupportAdaptive)
+        val child = replaceWithTransformerPlan(plan.child)
         logDebug(s"Columnar Processing for ${plan.getClass} is currently supported.")
         SortExecTransformer(plan.sortOrder, plan.global, child, plan.testSpillFrequency)
       case plan: ShuffleExchangeExec =>
         logDebug(s"Columnar Processing for ${plan.getClass} is currently supported.")
-        val child = replaceWithTransformerPlan(plan.child, isSupportAdaptive)
+        val child = replaceWithTransformerPlan(plan.child)
         if ((child.supportsColumnar || columnarConf.enablePreferColumnar) &&
           columnarConf.enableColumnarShuffle) {
           if (columnarConf.isVeloxBackend) {
             plan.outputPartitioning match {
               case HashPartitioning(exprs, _) =>
-                val projectChild = getProjectWithHash(exprs, child, isSupportAdaptive)
+                val projectChild = getProjectWithHash(exprs, child)
                 if (projectChild.supportsColumnar) {
-                  if (isSupportAdaptive) {
-                    ColumnarShuffleExchangeAdaptor(plan.outputPartitioning, projectChild,
-                      removeHashColumn = true)
+                  if (supportAdaptive(plan)) {
+                    ColumnarShuffleExchangeAdaptor(
```

Comment:
@FelixYBW If adding `CoalesceBatchesExec` here, below error will be triggered.

```
org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanExec.newQueryStage(AdaptiveSparkPlanExec.scala:537)
  at org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanExec.createQueryStages(AdaptiveSparkPlanExec.scala:489)
  at org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanExec.$anonfun$createQueryStages$2(AdaptiveSparkPlanExec.scala:519)
```
The code in Spark is as below, which expects `newShuffle.isInstanceOf[ShuffleExchangeLike]` but `CoalesceBatchesExec` is not.
```
  private def newQueryStage(e: Exchange): QueryStageExec = {
    val optimizedPlan = optimizeQueryStage(e.child, isFinalStage = false)
    val queryStage = e match {
      case s: ShuffleExchangeLike =>
        val newShuffle = applyPhysicalRules(
          s.withNewChildren(Seq(optimizedPlan)),
          postStageCreationRules(outputsColumnar = s.supportsColumnar),
          Some((planChangeLogger, "AQE Post Stage Creation")))
        if (!newShuffle.isInstanceOf[ShuffleExchangeLike]) {
          throw new IllegalStateException(
            "Custom columnar rules cannot transform shuffle node to something else.")
        }
```

Reply:
I see. Then let's put it into Velox in future.